### ISSUE TITLE
Port twisted.internet._win32stdio to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8018]
|<img alt="Doadin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Doadin reported|
|-|-|
|Trac ID|trac#8018|
|Type|defect|
|Created|2015-09-08 07:35:06Z|

Attachments:

* [[PATCH] Update _pollingfile.py for zope.interface python3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/96f/96feefe13f5d0b9316647556f8c02c3bacad1582/e23f62b1ffe6aa0c6a8c077f40559d7ad10e90d5.patch) (1240 bytes) - added by Doadin on 2015-09-08 07:35:33Z - 
* [[PATCH] Update _win32stdio.py for zope.interface python3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/96f/96feefe13f5d0b9316647556f8c02c3bacad1582/4fa8a274c424cbd3bb5488b87b201c566a0e3b4a.patch) (1075 bytes) - added by Doadin on 2015-09-08 07:35:46Z - 
* [[PATCH] [#8018] Update _win32stdio.py and _pollingfile.py for zope.interface python3 compatibility.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/96f/96feefe13f5d0b9316647556f8c02c3bacad1582/f9bb7d963ba4931b8964fb1fc63534ec881de811.patch) (2704 bytes) - added by Doadin on 2015-09-09 00:51:26Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__8018 8018
type__defect defect
reporter__Doadin Doadin
priority__high high
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__core core
keywords__ 
time__1441697706372944 1441697706372944
changetime__1467854739889221 1467854739889221
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

Hi Doadin,

Thanks for your patches.

Please note that the patches should be submitted as a single file together with test

More info about the Twisted dev process can be found here http://twistedmatrix.com/trac/wiki/TwistedDevelopment

See here for a patch which ports some code to py3

https://github.com/twisted/twisted/commit/59daefaa2a5363acfa9b3b049623458402529eae

Note that the ported modules and tests should be recorded in the dist3.py file.

Please let us know if you need any help with the patch.

Thanks!

**Comment by twisted-trac** [comment]
|<img alt="Doadin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|Doadin commented|
|-|-|

I updated the patch to be all in one and added a missing import. I think i might need help or someone to review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

This ticket needs a better description.

Why you need to update those file for python3 compatiblity?

What is the goal of this update?

What was not previously working and will work after this update?

This questions need to be answered in the description of this ticket.

The new functionality should be advertised in a news file fragment, submitted in the same path, so that the release notes will contain a note about this change.

More info about the news file can be found here http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set owner to @hawkowl|
|-|-|

This isn't a regression (we've never supported Py3 on Windows).

Once we get a Python 3 builder on Windows we can put run the tests and move this forward (hopefully next week?).

I'll assign it to me, as it's blocked on me getting that going.

**Comment by twisted-trac** [comment]
|<img alt="Doadin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|Doadin commented|
|-|-|

Any update on this? Its been 7 months.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl commented|
|-|-|

@Doadin I made some progress on this the other day -- see [#8025](https://github.com/twisted/twisted/issues/8025) and the precursor https://tm.tl/[[#8266](https://github.com/twisted/twisted/issues/8266)](https://github.com/twisted/twisted/issues/8266) for two tickets that would solve this in due course. If you have some spare time, a review on #8266 would be very appreciated :)

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_9"></a>|@rodrigc set status to `closed`|
|-|-|

Duplicate of [ticket:8291]
